src/shared/proto/radio_primitive.proto

```diff
@@ -0,0 +1,110 @@
+syntax = "proto3";
+
+// We call this a RadioPrimitive and not just a "Primitive"
+// because the contents of this message are a 1-1 mapping
+// with the existing `RadioPrimitive` we send over radio
+// using the MRF-Dongle.
+message RadioPrimitive
+{
+    // A numeric ID representing the primitive for firmware
+    FirmwarePrimitiveType prim_type = 1;
+
+    // the parameters to be encoded into the packet
+    double parameter1 = 2;
+    double parameter2 = 3;
+    double parameter3 = 4;
+    double parameter4 = 5;
+
+    // Extra bits used for flags and/or additional information
+    uint32 extra_bits = 6;
+
+    // Indicates whether the robot should move slowly (<1.5 m/s)
+    bool slow = 7;
+}
+
+/*
+ * This enum represents the IDs of the primitives implemented in firmware.
+ *
+ * TODO remove duplication w/ firmware_primitive_type.h and radio msgs once radio backend
+ * is on proto https://github.com/UBC-Thunderbots/Software/issues/1216
+ *
+ */
+enum FirmwarePrimitiveType
+{
+    /**
+     * Stop Primitive
+     *
+     * The parameters are unused.
+     *
+     * The extra field is 0 for coasting or 1 for braking.
+     */
+    STOP = 0;
+
+    /**
+     * Move Primitive
+     *
+     * The parameters are the relative position, the relative orientation,
+     * and the time delta.
+     *
+     * The extra field is 0 if the caller doesn’t care about orientation,
+     * or 1 if it does.
+     */
+    MOVE = 1;
+
+    /**
+     * Dribble Primitive
+     *
+     * The parameters are the relative position and orientation.
+     *
+     * The extra field is 0 if small kicks are prohibited or 1 if they are
+     * allowed.
+     */
+    DRIBBLE = 2;
+
+    /**
+     * Shoot Primitive
+     *
+     * Note that in HL, this is split into kick and chip.
+     *
+     * The parameters are the relative position, relative orientation, and
+     * power (either m/s or m).
+     *
+     * The extra field has bit 0 clear to kick or set to chip, and bit 1
+     * set if the caller cares about orientation.
+     */
+    SHOOT = 3;
+
+    /**
+     * Catch Primitive
+     *
+     * The parameters are the velocity, dribble speed, and margin.
```

Comment:
Would it be better if units were added to the primitives? Similar to how Shoot primitive mentioned `power (either m/s or m).

Reply:
good point, added units to all of them